connection.go

```diff
@@ -1666,3 +1719,52 @@ func (conn *Connection) ServerProtocolInfo() ProtocolInfo {
 func (conn *Connection) ClientProtocolInfo() ProtocolInfo {
 	return clientProtocolInfo.Clone()
 }
+
+func shutdownEventCallback(event WatchEvent) {
+	// Receives "true" on server shutdown.
+	// See https://www.tarantool.io/en/doc/latest/dev_guide/internals/iproto/graceful_shutdown/
+	// step 2.
+	val, ok := event.Value.(bool)
+	if ok && val {
+		// defer cause otherwise we'll block ourselves on ack wait.
+		defer event.Conn.processShutdown()
```

Comment:
Or will something change?
```suggestion
		event.Conn.processShutdown()
```

Reply:
Yes, it may lock the ack request and everything is doomed.

Reply:
Sorry, it seems that this one had left from the time I've had deadlock issues with unregister. Removed.